detox/src/devices/drivers/android/tools/FreeDeviceFinderBase.js

```diff
@@ -0,0 +1,40 @@
+const log = require('../../../../utils/logger').child({ __filename });
+
+const DEVICE_LOOKUP_LOG_EVT = 'DEVICE_LOOKUP';
+
+class FreeDeviceFinderBase {
+  constructor(adb, deviceRegistry) {
+    this.adb = adb;
+    this.deviceRegistry = deviceRegistry;
+  }
+
+  async findFreeDevice(deviceQuery) {
+    const { devices } = await this.adb.devices();
+    for (const candidate of devices) {
+      const isBusy = this.deviceRegistry.isDeviceBusy(candidate.adbName);
+      if (isBusy) {
+        log.debug({ event: DEVICE_LOOKUP_LOG_EVT }, `Device ${candidate.adbName} is busy, skipping...`);
+        continue;
+      }
+
```

Comment:
@d4vidi, while I was reviewing unit tests and the candidate class implementation, which is `DeviceHandle` class (or `EmulatorHandle` subclass), I've been struck by a question:

**Why don't we check that `candidate.status === 'device'` or at least `candidate.status !== 'offline'`?** Does not that makes some sense and adds a bit of safety for the edge-cases?

Reply:
Added a check for 'offline' devices